[Flang][Driver] NFC: Move Flang/Fortran only options to a separate file (#194398)
llvm/llvm-project: flang/docs/FlangDriver.md

Reviewer: ```suggestion
* add a completely new definition if the option that you are adding has not
been defined yet. These must be added to `clang/Options/FlangOptions.td`
unless they are intended to be shared with Clang, in which case they should
be added to `clang/Options/Options.td`
```

* leverage Clang's ability to drive various backends available in LLVM, as well
  as linkers and assemblers.
One implication of this dependency on Clang is that all of Flang's compiler
options are defined inside the Clang subproject. Flang-only options are in

> ``clang/include/clang/Options/FlangOptions.td`. For options that are common for both

Reviewer: extra backtick

```suggestion
`clang/include/clang/Options/FlangOptions.td`. For options that are common for both
```

Author: Done

> Flang and Clang, the corresponding definitions are shared in

Reviewer: ```suggestion
``clang/include/clang/Options/FlangOptions.td`. Options that are common to both
Flang and Clang are defined in
```

Author: Done

### Option Definition
All of Flang's compiler and frontend driver options are defined in
`clang/include/clang/Options/FlangOptions.td` and `clang/include/clang/Options/Options.td`.

> When adding a new option to Flang, you will either:

Reviewer: ```suggestion
When adding a new option to Flang, you will do one of the following:
```

  * extend the existing definition for an option that is already available

>     in one of Clang's drivers (e.g.  `clang`), but not yet available in Flang, or

Reviewer: ```suggestion
* extend the existing definition for an option that is already available
in one of Clang's drivers (e.g.  `clang`), but not yet available in Flang.
These options will be in `clang/Options/Options.td`.
```

### Option Definition
All of Flang's compiler and frontend driver options are defined in
`clang/include/clang/Options/FlangOptions.td` and `clang/include/clang/Options/Options.td`

> in Clang. When adding a new option to Flang, you will either:

Reviewer: Nit.

```suggestion
`clang/include/clang/Options/FlangOptions.td` and `clang/include/clang/Options/Options.td`.
When adding a new option to Flang, you will either:
```

Author: Done
